lineapy/plugins/argo_pipeline_writer.py

```diff
@@ -0,0 +1,255 @@
+)
+from lineapy.plugins.utils import load_plugin_template
+from lineapy.utils.logging_config import configure_logging
+from lineapy.utils.utils import prettify
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+
+class ARGODagFlavor(Enum):
+    StepPerSession = 1
+    StepPerArtifact = 2
+
+
+ARGODAGConfig = TypedDict(
+    "ARGODAGConfig",
+    {
+        "namespace": str,
+        "host": str,
+        "verify_ssl": str,
+        "workflow_name": str,
+        "service_account": int,
+        "kube_config": str,
+        "task_serialization": str,
+        "dag_flavor": str,
+    },
+    total=False,
+)
+
+
+class ARGOPipelineWriter(BasePipelineWriter):
+    """
+    Class for pipeline file writer. Corresponds to "ARGO" framework.
+    """
+
+    @property
+    def docker_template_name(self) -> str:
+        return "argo_dockerfile.jinja"
+
+    def _write_dag(self) -> None:
+
+        # Check if the given DAG flavor is a supported/valid one
+        try:
+            dag_flavor = ARGODagFlavor[
+                self.dag_config.get("dag_flavor", "StepPerSession")
+            ]
+        except KeyError:
+            raise ValueError(f'"{dag_flavor}" is an invalid ARGO dag flavor.')
+
+        # Construct DAG text for the given flavor
+        full_code = self._write_operators(dag_flavor)
+
+        # Write out file
+        file = self.output_dir / f"{self.pipeline_name}_dag.py"
+        file.write_text(prettify(full_code))
+        logger.info(f"Generated DAG file: {file}")
+
+    def _write_operators(
+        self,
+        dag_flavor: ARGODagFlavor,
+    ) -> str:
+
+        DAG_TEMPLATE = load_plugin_template("argo_dag.jinja")
+
+        if dag_flavor == ARGODagFlavor.StepPerSession:
+            task_breakdown = DagTaskBreakdown.TaskPerSession
+        elif dag_flavor == ARGODagFlavor.StepPerArtifact:
+            task_breakdown = DagTaskBreakdown.TaskPerArtifact
+
+        # Get task definitions based on dag_flavor
+        task_defs, task_graph = get_task_graph(
+            self.artifact_collection,
+            pipeline_name=self.pipeline_name,
+            task_breakdown=task_breakdown,
+        )
+
+        task_defs["setup"] = get_noop_setup_task_definition(self.pipeline_name)
+        task_defs["teardown"] = get_noop_teardown_task_definition(
+            self.pipeline_name
+        )
+        # insert in order to task_names so that setup runs first and teardown runs last
+        task_graph.insert_setup_task("setup")
+        task_graph.insert_teardown_task("teardown")
+
+        task_names = list(task_defs.keys())
+
+        task_defs = {tn: task_defs[tn] for tn in task_names}
+
+        (
+            rendered_task_defs,
+            task_loading_blocks,
+        ) = self.get_rendered_task_definitions(task_defs)
+
+        # Handle dependencies
+        task_dependencies = reversed(
+            [f"{task0} >> {task1}" for task0, task1 in task_graph.graph.edges]
+        )
+
+        return_artifacts = []
```

Comment:
Is this actually being used by the template?

Reply:
Removed :+1: